e2e-tests/self-upgrade-no-symlink.sh

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+unset CARGO_INSTALL_ROOT
+
+export CARGO_HOME=$(mktemp -d 2>/dev/null || mktemp -d -t 'cargo-home')
+export PATH="$CARGO_HOME/bin:$PATH"
+
+# first boostrap-install into the CARGO_HOME
+"./$1" binstall --no-confirm --no-symlinks --force cargo-binstall
+
+# now we're running the CARGO_HOME/bin/cargo-binstall (via cargo):
+
+# self update
+cargo binstall --no-confirm --no-symlinks --force cargo-binstall
```

Comment:
There it is!

Reply:
Actually, we could work around that by manually copying `./$1` into `$CARGO_HOME/bin`.
Let me try that.

Reply:
Yes, that also causes the CI to fail as expected!